pkg/apis/core/helper/helpers.go

```diff
@@ -139,6 +139,9 @@ var podComputeQuotaResources = sets.NewString(
 	string(core.ResourceLimitsMemory),
 	string(core.ResourceRequestsCPU),
 	string(core.ResourceRequestsMemory),
+	string(core.ResourceEphemeralStorage),
+	string(core.ResourceRequestsEphemeralStorage),
+	string(core.ResourceLimitsEphemeralStorage),
 )
```

Comment:
Changing `podComputeQuotaResources` modifies the behavior of `IsResourceQuotaScopeValidForResource`, which is used in API validation.

A good first step is to determine who is using this list and this method, and limit the impact. It looks like we have this call tree:

```
helpers.podComputeQuotaResources
	helpers.IsResourceQuotaScopeValidForResource
		validation.validateResourceQuotaScopes
		validation.validateScopedResourceSelectorRequirement
```

To limit impact across packages, before making any changes to the list, move the existing `podComputeQuotaResources` and `IsResourceQuotaScopeValidForResource` methods to the validation package and make them both private.

Reply:
Secondly, because this is relaxing validation, this has to be rolled out over two releases, so that a cluster running a 1.23 and 1.22 API server together will not allow data to be persisted via 1.23 that a 1.22 API server would consider invalid.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-enum-value-in-existing-field is an example of how to relax validation over two releases (it is describing relaxing an enum value, but the same approach works for relaxing the allowed values here).

It looks like resource quota validation already constructs and plumbs `ResourceQuotaValidationOptions` to `validateResourceQuotaScopes` and `validateScopedResourceSelectorRequirement`, which is great... that means a lot of the plumbing work is already done.

In 1.23, you would need to:
1. add an `AllowEphemeralStorageInScopedQuota bool` option to `ResourceQuotaValidationOptions` (defaulting to false)
2. modify `getValidationOptionsFromResourceQuota` to set `AllowEphemeralStorageInScopedQuota` to true if and only if the new ephemeral storage values you want to allow are already in use in the existing quota object (so it would be false for create requests, and conditionally true for update requests if the existing object is already setting this data)
3. plumb `ResourceQuotaValidationOptions` down into the (newly private) `isResourceQuotaScopeValidForResource` method
4. in `isResourceQuotaScopeValidForResource`, only allow the expanded values if `opts.AllowEphemeralStorageInScopedQuota` is true
5. Add unit tests exercising:
   * create (new value is disallowed)
   * update of an object that doesn't have the new value that tries to set the new value (new value is disallowed)
   * update of an object that already has the new value (new value is allowed)

In the next release, you could unconditionally set `AllowEphemeralStorageInScopedQuota` to true (or remove `AllowEphemeralStorageInScopedQuota` entirely and just make `isResourceQuotaScopeValidForResource` use the expanded list to validate)

Reply:
> Changing `podComputeQuotaResources` modifies the behavior of `IsResourceQuotaScopeValidForResource`, which is used in API validation.
> 
> A good first step is to determine who is using this list and this method, and limit the impact. It looks like we have this call tree:
> 
> ```
> helpers.podComputeQuotaResources
> 	helpers.IsResourceQuotaScopeValidForResource
> 		validation.validateResourceQuotaScopes
> 		validation.validateScopedResourceSelectorRequirement
> ```
> 
> To limit impact across packages, before making any changes to the list, move the existing `podComputeQuotaResources` and `IsResourceQuotaScopeValidForResource` methods to the validation package and make them both private.

If we move the existing podComputeQuotaResources and IsResourceQuotaScopeValidForResource methods to the validation package，we will break the third-party library which depend on this method, should we just copy it？

Reply:
done, PTAL